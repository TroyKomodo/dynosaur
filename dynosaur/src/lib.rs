#![doc = include_str!("../README.md")]

pub use dynosaur_derive::*;
