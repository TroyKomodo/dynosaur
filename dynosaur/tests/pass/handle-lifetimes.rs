use dynosaur::dynosaur;

#[dynosaur(DynMyTrait)]
trait MyTrait {
    type Item;
    async fn foo<T>(&self, x: &T) -> i32;
}

fn main() {}
