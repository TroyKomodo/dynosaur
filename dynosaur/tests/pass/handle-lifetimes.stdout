#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2021::*;
#[macro_use]
extern crate std;
use dynosaur::dynosaur;

trait MyTrait {
    type Item;
    async fn foo<T>(&self, x: &T)
    -> i32;
}
trait ErasedMyTrait {
    type Item;
    fn foo<'life0, 'life1, 'dynosaur, T>(&'life0 self, x: &'life1 T)
    ->
        ::core::pin::Pin<Box<dyn ::core::future::Future<Output = i32> +
        'dynosaur>>
    where
    T: 'dynosaur,
    'life0: 'dynosaur,
    'life1: 'dynosaur,
    Self: 'dynosaur;
}

fn main() {}
